# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/rustc/sparc.patch
# Copyright (C) 2023 - 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- rustc-1.69.0-src/vendor/object/src/write/elf/object.rs.vanilla	2023-05-04 20:43:01.836676712 +0200
+++ rustc-1.69.0-src/vendor/object/src/write/elf/object.rs	2023-05-04 20:44:42.775682447 +0200
@@ -85,6 +85,7 @@
             Architecture::S390x => true,
             Architecture::Sbf => false,
             Architecture::Sharc => true,
+            Architecture::Sparc => true,
             Architecture::Sparc64 => true,
             Architecture::Xtensa => true,
             _ => {
@@ -285,6 +286,7 @@
             (Architecture::S390x, None) => elf::EM_S390,
             (Architecture::Sbf, None) => elf::EM_SBF,
             (Architecture::Sharc, None) => elf::EM_SHARC,
+            (Architecture::Sparc, None) => elf::EM_SPARC,
             (Architecture::Sparc64, None) => elf::EM_SPARCV9,
             (Architecture::Xtensa, None) => elf::EM_XTENSA,
             _ => {
--- rustc-1.69.0-src/vendor/object/src/common.rs.vanilla	2023-05-04 20:28:36.842627565 +0200
+++ rustc-1.69.0-src/vendor/object/src/common.rs	2023-05-04 20:32:28.876640748 +0200
@@ -25,6 +25,7 @@
     S390x,
     Sbf,
     Sharc,
+    Sparc,
     Sparc64,
     Wasm32,
     Xtensa,
@@ -56,6 +57,7 @@
             Architecture::S390x => Some(AddressSize::U64),
             Architecture::Sbf => Some(AddressSize::U64),
             Architecture::Sharc => Some(AddressSize::U32),
+            Architecture::Sparc => Some(AddressSize::U32),
             Architecture::Sparc64 => Some(AddressSize::U64),
             Architecture::Wasm32 => Some(AddressSize::U32),
             Architecture::Xtensa => Some(AddressSize::U32),
--- rustc-1.78.0-src/vendor/object-0.32.2/src/write/elf/object.rs.vanilla	2024-05-18 18:52:02.723910857 +0200
+++ rustc-1.78.0-src/vendor/object-0.32.2/src/write/elf/object.rs	2024-05-18 18:53:46.458905632 +0200
@@ -142,6 +142,7 @@
             Architecture::S390x => true,
             Architecture::Sbf => false,
             Architecture::Sharc => true,
+            Architecture::Sparc => true,
             Architecture::Sparc64 => true,
             Architecture::Xtensa => true,
             _ => {
@@ -303,6 +304,7 @@
             (Architecture::S390x, None) => elf::EM_S390,
             (Architecture::Sbf, None) => elf::EM_SBF,
             (Architecture::Sharc, None) => elf::EM_SHARC,
+            (Architecture::Sparc, None) => elf::EM_SPARC,
             (Architecture::Sparc64, None) => elf::EM_SPARCV9,
             (Architecture::Xtensa, None) => elf::EM_XTENSA,
             _ => {
--- rustc-1.78.0-src/vendor/object-0.32.2/src/common.rs.vanilla	2024-05-18 19:02:28.062879358 +0200
+++ rustc-1.78.0-src/vendor/object-0.32.2/src/common.rs	2024-05-18 19:03:31.611876157 +0200
@@ -27,6 +27,7 @@
     S390x,
     Sbf,
     Sharc,
+    Sparc,
     Sparc64,
     Wasm32,
     Wasm64,
@@ -70,6 +71,7 @@
             Architecture::S390x => Some(AddressSize::U64),
             Architecture::Sbf => Some(AddressSize::U64),
             Architecture::Sharc => Some(AddressSize::U32),
+            Architecture::Sparc => Some(AddressSize::U32),
             Architecture::Sparc64 => Some(AddressSize::U64),
             Architecture::Wasm32 => Some(AddressSize::U32),
             Architecture::Wasm64 => Some(AddressSize::U64),
--- rustc-1.78.0-src/vendor/object-0.32.0/src/write/elf/object.rs.vanilla	2024-05-18 19:06:22.171867566 +0200
+++ rustc-1.78.0-src/vendor/object-0.32.0/src/write/elf/object.rs	2024-05-18 19:06:40.441866646 +0200
@@ -141,6 +141,7 @@
             Architecture::Riscv32 => true,
             Architecture::S390x => true,
             Architecture::Sbf => false,
+            Architecture::Sparc => true,
             Architecture::Sparc64 => true,
             Architecture::Xtensa => true,
             _ => {
@@ -345,6 +346,7 @@
             Architecture::Riscv64 => elf::EM_RISCV,
             Architecture::S390x => elf::EM_S390,
             Architecture::Sbf => elf::EM_SBF,
+            Architecture::Sparc => elf::EM_SPARC,
             Architecture::Sparc64 => elf::EM_SPARCV9,
             Architecture::Xtensa => elf::EM_XTENSA,
             _ => {
--- rustc-1.78.0-src/vendor/object-0.32.0/src/common.rs.vanilla	2024-05-18 19:06:50.191866155 +0200
+++ rustc-1.78.0-src/vendor/object-0.32.0/src/common.rs	2024-05-18 19:07:05.445865386 +0200
@@ -26,6 +26,7 @@
     Riscv64,
     S390x,
     Sbf,
+    Sparc,
     Sparc64,
     Wasm32,
     Wasm64,
@@ -59,6 +60,7 @@
             Architecture::Riscv64 => Some(AddressSize::U64),
             Architecture::S390x => Some(AddressSize::U64),
             Architecture::Sbf => Some(AddressSize::U64),
+            Architecture::Sparc => Some(AddressSize::U32),
             Architecture::Sparc64 => Some(AddressSize::U64),
             Architecture::Wasm32 => Some(AddressSize::U32),
             Architecture::Wasm64 => Some(AddressSize::U64),
--- rustc-1.78.0-src/compiler/rustc_codegen_ssa/src/back/metadata.rs.vanilla	2024-05-18 18:49:00.480920037 +0200
+++ rustc-1.78.0-src/compiler/rustc_codegen_ssa/src/back/metadata.rs	2024-05-18 18:48:28.853921630 +0200
@@ -208,6 +208,7 @@
         "powerpc64" => (Architecture::PowerPc64, None),
         "riscv32" => (Architecture::Riscv32, None),
         "riscv64" => (Architecture::Riscv64, None),
+        "sparc" => (Architecture::Sparc, None),
         "sparc64" => (Architecture::Sparc64, None),
         "avr" => (Architecture::Avr, None),
         "msp430" => (Architecture::Msp430, None),
